py-geopolars/src/lib.rs

```diff
@@ -22,9 +22,29 @@ fn convex_hull(series: &PyAny) -> PyResult<PyObject> {
     ffi::rust_series_to_py_series(&out)
 }
 
+#[pyfunction]
+fn euclidean_length(series: &PyAny) -> PyResult<PyObject> {
+    let series = ffi::py_series_to_rust_series(series)?;
+    let out = series
+        .euclidean_length()
+        .map_err(|e| PyValueError::new_err(format!("Something went wrong: {:?}", e)))?;
+    ffi::rust_series_to_py_series(&out)
+}
+
+#[pyfunction]
+fn geodesic_length(series: &PyAny) -> PyResult<PyObject> {
+    let series = ffi::py_series_to_rust_series(series)?;
+    let out = series
+        .geodesic_length(GeodesicLengthMethod::Geodesic)
+        .map_err(|e| PyValueError::new_err(format!("Something went wrong: {:?}", e)))?;
+    ffi::rust_series_to_py_series(&out)
+}
+
```

Comment:
@kylebarron Not too familiar with py03 and how it does arguments to functions. Whats the best way to pass through an optional named parameter for method here? 

We want the function signature to be series.geodesic_length(method="Geodesic")  but have series.geodesic_length() to default to method = "Geodesic" right?

Do we pass it in as an Option<PyObject> and transom in to a string which we match on?

Reply:
I think you could either do the default on the Rust side, by taking the input to be `Option<String>` or `Option<Method>` for some method enum. Or you could do the default on the Python side via https://pyo3.rs/v0.16.4/class.html#method-arguments (I haven't tried this myself).

Not sure which is preferable. Maybe good to look at py-polars to see how they do it.